UTILITIES/OFLS_File_Utilities.py

```diff
@@ -0,0 +1,177 @@
+    hist_dict['dither'] = split_status_line[6]
+    
+    # Done with the OFLS file - close it
+    ofls_file.close()
+    
+    # Return the dictionary
+    return hist_dict
+
+#--------------------------------------------------------------------
+#
+# Function - Read_ACIS_Continuity_file
+#
+#--------------------------------------------------------------------
+def Read_ACIS_Continuity_file(ofls_dir):
+    """
+    A basic file which will read the contents of an ACIS-Continuity.txt file
+    and return the information in a dictionary
+
+    input: ofls_dir - Path to the OFLS directory
+
+    output: continuity_dict - Dictionary containing the information stored
+                              in the ACIS-Continuity.txt file
+    """
+    # Open the ACIS-Continuity.txt file in the OFLS directory for reading
+    cont_file = open( '/'.join((ofls_dir, 'ACIS-Continuity.txt')), 'r')
+    
+    # Read the first line which supplies the path to the Continuity file
+    cont_path = cont_file.readline()[:-1]
+    
+    continuity_dict = {'cont_load_path': cont_path}
+ 
+    # Read the second line which tells you if the Review load is Normal
+    # in which case you will see only the word "Normal" in the line,
+    # or if the Review load is a TOO, SCS-107, or Full Stop load in which
+    # case you will see the type, and then the time of interrupt of the
+    # Continuity load.
+    load_type = cont_file.readline()
+
+    # Done with the file - close it
+    cont_file.close()
+
+    # Split the second line on spaces
+    splitline = load_type.split()
+
+    # Store the type in the dictionary no matter which it is
+    continuity_dict['type'] = splitline[0]
+
+    # If the length of the list is one, it's a Normal load
+    if len(splitline) == 1:
+        # ...and the interrupt time will be set to None
+        continuity_dict['interrupt_date'] = None
+        continuity_dict['interrupt_time'] = None
+    else:  # Else the load type is one of the three interrupt types
+         continuity_dict['interrupt_date'] = splitline[1]
+         continuity_dict['interrupt_time'] = apt.secs(splitline[1])
+
+    # Finished capturing the information - return the dictionary
+    return continuity_dict
+
+
+
+#--------------------------------------------------------------------
+#
+# Function - Get_Continuity_End_Status
+#
+#--------------------------------------------------------------------
+def Get_OFLS_Status_Line(ofls_dir):
+    """
+    Given an OFLS directory this function will obtain the status line 
+    at either the end of the Continuity load, or at the point where the
+    Continuity load was cut due to a TOO, SCS-107 or Full Stop.
+
+    input: ofls_dir - path to the OFLS directory of the Review load
+
+    output: continuity_status - Dictionary of the 
+    """
+    # If an ACIS-History_edit.txt file exists in the Review OFLS directory
+    # then you can use that to capture the status data at the cut time of the
```

Comment:
Thinking of the event like the failed grating move here... is the status grabbed here one we supply at an interrupt like history-files, or is it the last status before the interrupt time?  If the latter, we might have problems with recovery loads from such hardware mishaps.

Reply:
It's the one we supply when we run history-files.pl